install_scripts/templates/common/ip-address.sh

```diff
@@ -109,6 +109,24 @@ discoverPublicIp() {
         printf "The installer will use service address '%s' (discovered from EC2 metadata service)\n" "$PUBLIC_ADDRESS"
         return
     fi
+
+    # azure
+    if commandExists "curl"; then
+        set +e
+        _out=$(curl --noproxy "*" --max-time 5 --connect-timeout 2 -qSfs -H Metadata:true "http://169.254.169.254/metadata/instance/network/interface/0/ipv4/ipAddress/0/publicIpAddress?api-version=2017-08-01&format=text" 2>/dev/null)
```

Comment:
I don't think you'd need to redirect stderr to /dev/null if you removed the `S` flag from curl

Reply:
I copied the code from the AWS/GCE handlers - only difference is the URL